### PR TITLE
Alter table crash in antlr parser unsupported feature

### PR DESCRIPTION
**Pull request by Jakeowen1**
### Description

1. Currently Babelfish crashes when alter table alter column is called with drop default where the column has no definition
2. This change was made to prevent customers instances from crashing when calling this query

### Issues Resolved

[Babel-3818]

### Test Scenarios Covered ###
* **Use case based -**

```
1> alter table cities alter column region drop default defRegion
2> go
Msg 33557097, Level 16, State 1, Server BABELFISH, Line 1
'DROP DEFAULT' is not currently supported in Babelfish
```

```
1> alter table cities alter column region varchar(50) drop default defRegion
2> go
Msg 33557097, Level 16, State 1, Server BABELFISH, Line 1
'DROP DEFAULT' is not currently supported in Babelfish
```

```
1> alter table cities alter region varchar(50) drop default defRegion
2> go
Msg 33557097, Level 16, State 1, Server BABELFISH, Line 1
syntax error near 'region' at line 1 and character position 25
```

```
1> alter table cities alter region drop default defRegion
2> go
Msg 33557097, Level 16, State 1, Server BABELFISH, Line 1
syntax error near 'region' at line 1 and character position 25
```

```
1> alter table cities alter region varchar(50) drop default
2> go
Msg 33557097, Level 16, State 1, Server BABELFISH, Line 1
syntax error near 'region' at line 1 and character position 25
```

* **Arbitrary inputs -**

- Column not defined in table
```
1> alter table cities alter column fake_column drop default defRegion
2> go
Msg 33557097, Level 16, State 1, Server BABELFISH, Line 1
'DROP DEFAULT' is not currently supported in Babelfish
```

* **Minor version upgrade tests -**
Added to upgrade scripts

* **Major version upgrade tests -**
Added to upgrade scripts

* **Performance tests -**
NA

* **Tooling impact -**
NA

* **Client tests -**
Example tests used are reflective of client calls that caused error

### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/src/tsqlUnsupportedFeatureHandler.cpp [hunk 1]
```diff
@@ -737,16 +737,18 @@ antlrcpp::Any TsqlUnsupportedFeatureHandlerImpl::visitAlter_table(TSqlParser::Al
 
 	if (ctx->COLUMN()) // ALTER TABLE ... ALTER COLUMN
 	{
-		Assert(ctx->column_definition());
```
Comment (forestkeeper) [inline contrib/babelfishpg_tsql/src/tsqlUnsupportedFeatureHandler.cpp:740]: Can we still keep the assert in here ? I think this change is not related to null column_def

Comment (Jakeowen1) [reply contrib/babelfishpg_tsql/src/tsqlUnsupportedFeatureHandler.cpp:740]: The assert is what was causing the crash. When we call `alter table cities alter column region drop default defRegion` the antlr parser is expecting a column definition after "region" which is why there was no crash originally when running `alter table cities alter column region varchar(50) drop default defRegion`. Perhaps there is another way to handle the assert that could be better?

Comment (forestkeeper) [reply contrib/babelfishpg_tsql/src/tsqlUnsupportedFeatureHandler.cpp:740]: So for alter table cities alter column region drop default defRegion , the ctx->column_definition() = null, why I didn't find the if cdctx == null clause in if conditon

**Review by forestkeeper — APPROVED** [review]
LGTM
